lib/xandra/cluster/control_connection.ex

```diff
@@ -0,0 +1,119 @@
+defmodule Xandra.Cluster.ControlConnection do
+  use Connection
+
+  alias Xandra.{Frame, Protocol, Connection.Utils}
+
+  require Logger
+
+  @default_timeout 5_000
+  @socket_options [packet: :raw, mode: :binary, active: false]
+
+  defstruct [:address, :port, :cluster, :socket, new: true, buffer: <<>>]
+
+  def start_link(cluster, address, port) do
+    state = %__MODULE__{cluster: cluster, address: address, port: port}
+    Connection.start_link(__MODULE__, state)
+  end
+
+  def init(state) do
+    {:connect, :init, state}
+  end
+
+  def connect(_action, %__MODULE__{address: address, port: port} = state) do
+    case :gen_tcp.connect(address, port, @socket_options, @default_timeout) do
+      {:ok, socket} ->
+        state = %{state | socket: socket}
+        with {:ok, supported_options} <- Utils.request_options(socket),
+             :ok <- startup_connection(socket, supported_options),
+             :ok <- register_to_events(socket),
+             :ok <- :inet.setopts(socket, active: :true),
```

Comment:
This is dangerous and usually not recommended. I think we should go with `active: :once`, and then set `active: :once` again in `handle_info({:tcp, socket, data}, ...)`.